File: Offer/2224/main.py
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        root: ListNode = ListNode(0)
        temp: ListNode = None
        while head != None:
            temp = head.next
            head.next = root.next
            root.next = head
            head = temp
        return root.next

File: Offer/2224/main.swift
  public class ListNode {
      public var val: Int
      public var next: ListNode?
      public init() { self.val = 0; self.next = nil; }
      public init(_ val: Int) { self.val = val; self.next = nil; }
      public init(_ val: Int, _ next: ListNode?) { self.val = val; self.next = next; }
  }

class Solution {
    func reverseList(_ head: ListNode?) -> ListNode? {
        var head = head
        var root = ListNode()
        var temp: ListNode? = nil
        while head != nil{
            temp = head!.next
            head!.next = root.next
            root.next = head
            head = temp
        }
        return root.next
    }
}